src/blocks.zig

```diff
@@ -51,43 +51,190 @@ pub const Ore = struct {
+
+				std.mem.copyBackwards(DataType, self.data[(insertIdx + 1) .. self.allocatedSize + 1], self.data[insertIdx..self.allocatedSize]);
+
+				self.idxLookup[insertIdx] = blockId;
+				self.data[insertIdx] = propVal;
+
+				self.allocatedSize += 1;
+			}
+
+			pub fn clear(self: *Self) void {
+				self.allocatedSize = 0;
+			}
+		};
+	}
+}
+
+fn isSortedProp(comptime T: type) bool {
+	return std.meta.hasFn(T, "clear") and
+		std.meta.hasFn(T, "addBlockProperty") and
+		std.meta.hasFn(T, "getBlockPropertyValue");
+}
+
+fn resetSortedProperties() void {
+	inline for(@typeInfo(BlockProps).@"struct".decls) |decl| {
+		const sortedProp = &@field(BlockProps, decl.name);
+
+		if(comptime isSortedProp(@TypeOf(sortedProp.*))) {
+			std.log.info("Cleared \'{s}\' from {d} entries", .{decl.name, sortedProp.allocatedSize});
+			sortedProp.clear();
+		}
+	}
+}
+
+const BlockProps = struct {
+	pub var transparent: [maxBlockCount]bool = undefined;
+	pub var collide: [maxBlockCount]bool = undefined;
+	pub var id: [maxBlockCount][]u8 = undefined;
+
+	pub var blockHealth: [maxBlockCount]f32 = undefined;
+	pub var blockResistance: [maxBlockCount]f32 = undefined;
+
+	/// Whether you can replace it with another block, mainly used for fluids/gases
+	pub var replacable: [maxBlockCount]bool = undefined;
+	pub var selectable: [maxBlockCount]bool = undefined;
+	pub var blockDrops: [maxBlockCount][]BlockDrop = undefined;
+	/// Meaning undegradable parts of trees or other structures can grow through this block.
+	pub var degradable: [maxBlockCount]bool = undefined;
+	pub var viewThrough: [maxBlockCount]bool = undefined;
+	pub var alwaysViewThrough: [maxBlockCount]bool = undefined;
+	pub var hasBackFace: [maxBlockCount]bool = undefined;
+	pub var blockTags: [maxBlockCount][]Tag = undefined;
+	pub var light: [maxBlockCount]u32 = undefined;
+	/// How much light this block absorbs if it is transparent
+	pub var absorption: [maxBlockCount]u32 = undefined;
+	pub var mode: [maxBlockCount]*RotationMode = undefined;
+	pub var modeData: [maxBlockCount]u16 = undefined;
+	pub var lodReplacement: [maxBlockCount]u16 = undefined;
+	pub var opaqueVariant: [maxBlockCount]u16 = undefined;
+
+	pub var friction: [maxBlockCount]f32 = undefined;
+	pub var bounciness: [maxBlockCount]f32 = undefined;
+	pub var density: [maxBlockCount]f32 = undefined;
+	pub var terminalVelocity: [maxBlockCount]f32 = undefined;
+	pub var mobility: [maxBlockCount]f32 = undefined;
+
+	/// ------------------------------------------------- Sorted Block Properties
+	/// These properties are rarely used, so to save memory we use sorted arrays
+	/// with ~100-300 entries instead of creating arrays with maxBlockCount (65536) entries.
+	/// Magic Value - Increase it if you need to store more block properties.
+	/// Consider creating a separate variable with a comment for a specific property if only one needs a larger size.
+	pub const maxSortedBlockProperties: usize = 100;
```

Comment:
Magic value is unfortunately unacceptable, as you don't know how many of them will be added by add-ons. I don't think the scale would be so huge that falling to array storage makes sense, but it should at least be resizeable to meet the demand in worst case scenario.